examples/lifelong_learning/RFNet/basemodel.py

```diff
@@ -0,0 +1,390 @@
+
+    def train(self, train_data, valid_data=None, **kwargs):
+        self.trainer = Trainer(self.train_args, train_data=train_data)
+        print("Total epoches:", self.trainer.args.epochs)
+        for epoch in range(
+                self.trainer.args.start_epoch,
+                self.trainer.args.epochs):
+            if epoch == 0 and self.trainer.val_loader:
+                self.trainer.validation(epoch)
+            self.trainer.training(epoch)
+
+            if self.trainer.args.no_val and (
+                    epoch %
+                    self.trainer.args.eval_interval == (
+                        self.trainer.args.eval_interval -
+                        1) or epoch == self.trainer.args.epochs -
+                    1):
+                # save checkpoint when it meets eval_interval or the training
+                # finished
+                is_best = False
+                self.train_model_url = self.trainer.saver.save_checkpoint({
+                    'epoch': epoch + 1,
+                    'state_dict': self.trainer.model.state_dict(),
+                    'optimizer': self.trainer.optimizer.state_dict(),
+                    'best_pred': self.trainer.best_pred,
+                }, is_best)
+
+            # if not self.trainer.args.no_val and \
+            #         epoch % self.train_args.eval_interval == (self.train_args.eval_interval - 1) \
+            #         and self.trainer.val_loader:
+            #     self.trainer.validation(epoch)
+
+        self.trainer.writer.close()
+
+        return self.train_model_url
+
+    def predict(self, data, **kwargs):
+        if not isinstance(data[0][0], dict):
+            data = preprocess(data)
+
+        if isinstance(data, np.ndarray):
+            data = data.tolist()
+
+        self.validator.test_loader = DataLoader(
+            data,
+            batch_size=self.val_args.test_batch_size,
+            shuffle=False,
+            pin_memory=True)
+        return self.validator.validate()
+
+    def evaluate(self, data, **kwargs):
+        samples = preprocess(data.x)
+        predictions = self.predict(samples)
+        return accuracy(data.y, predictions)
+
+    def load(self, model_url, **kwargs):
+        if model_url:
+            self.validator.new_state_dict = torch.load(
+                model_url, map_location=torch.device("cpu"))
+            self.validator.model = load_my_state_dict(
+                self.validator.model, self.validator.new_state_dict['state_dict'])
+
+            self.train_args.resume = model_url
+        else:
+            raise Exception("model url does not exist.")
+
+    def save(self, model_path=None):
+        # TODO: save unstructured data model
+        if not model_path:
+            LOGGER.warning(f"Not specify model path.")
+            return self.train_model_url
+
+        return FileOps.upload(self.train_model_url, model_path)
+
+
+def train_args():
+    parser = argparse.ArgumentParser(description="PyTorch RFNet Training")
```

Comment:
@luosiqi 
The command-line parsing module argparse should not be used, because it dose not use in this scense. It's easy to misunderstand.

Reply:
@luosiqi suggest that:
```python
Class TrainArgs:   
    def __init__(self, **kwargs):   
         self.depth = kwargs.get('depth', False)
         self.dateaset = Context.get_parameters('dataset', 'cityscapes')
``